packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -446,17 +446,39 @@ function transform(codeGen: CodeGen): t.Expression {
         }
 
         // Properties
+        const propsObj = t.objectExpression([]);
+
+        // Properties: standard props
         if (props) {
-            const propsObj = objectToAST(props, (key) => computeAttrValue(props[key], element));
+            for (const [key, value] of Object.entries(props)) {
+                propsObj.properties.push(
+                    t.property(t.literal(key), computeAttrValue(value, element))
+                );
+            }
+        }
+
+        // Properties: lwc:inner-html directive
+        if (lwc?.innerHTML) {
+            const expr =
+                typeof lwc.innerHTML === 'string'
+                    ? t.literal(lwc.innerHTML)
+                    : bindExpression(lwc.innerHTML, element);
+
+            propsObj.properties.push(
+                t.property(t.identifier('innerHTML'), codeGen.genSanitizedHtmlExpr(expr))
```

Comment:
What is the difference between using `t.identifier('innerHTML')` and `t.literal('innerHTML')` here? I tried it out locally but it seems to produce the same output.